bin/ruby-build

```diff
@@ -206,6 +206,21 @@ is_freebsd() {
   [ "$(uname -s)" = "FreeBSD" ]
 }
 
+is_fedora() {
+  if [ -r /etc/os-release ]; then
```

Comment:
If `/etc/os-release` doesn't exist or is not readable, `is_fedora` would exit with 0 (success status). Let's change to an early return if the file is not readable:
```suggestion
  [ -r /etc/os-release ] || return 1
```